app/src/main/java/com/keylesspalace/tusky/TabData.kt

```diff
@@ -29,20 +29,22 @@ const val NOTIFICATIONS = "Notifications"
 const val LOCAL = "Local"
 const val FEDERATED = "Federated"
 const val DIRECT = "Direct"
+const val HASHTAG = "Hashtag"
 
 data class TabData(val id: String,
                    @StringRes val text: Int,
                    @DrawableRes val icon: Int,
-                   val fragment: () -> Fragment)
+                   val fragment: (List<String>) -> Fragment,
+                   val arguments: List<String> = emptyList())
 
-
-fun createTabDataFromId(id: String): TabData {
+fun createTabDataFromId(id: String, arguments: List<String> = emptyList()): TabData {
     return when (id) {
-        HOME -> TabData(HOME, R.string.title_home, R.drawable.ic_home_24dp) { TimelineFragment.newInstance(TimelineFragment.Kind.HOME) }
-        NOTIFICATIONS -> TabData(NOTIFICATIONS, R.string.title_notifications, R.drawable.ic_notifications_24dp) { NotificationsFragment.newInstance() }
-        LOCAL -> TabData(LOCAL, R.string.title_public_local, R.drawable.ic_local_24dp) { TimelineFragment.newInstance(TimelineFragment.Kind.PUBLIC_LOCAL) }
-        FEDERATED -> TabData(FEDERATED, R.string.title_public_federated, R.drawable.ic_public_24dp) { TimelineFragment.newInstance(TimelineFragment.Kind.PUBLIC_FEDERATED) }
-        DIRECT -> TabData(DIRECT, R.string.title_direct_messages, R.drawable.reblog_direct_dark) { ConversationsFragment.newInstance() }
+        HOME -> TabData(HOME, R.string.title_home, R.drawable.ic_home_24dp, { TimelineFragment.newInstance(TimelineFragment.Kind.HOME) })
+        NOTIFICATIONS -> TabData(NOTIFICATIONS, R.string.title_notifications, R.drawable.ic_notifications_24dp, { NotificationsFragment.newInstance() })
+        LOCAL -> TabData(LOCAL, R.string.title_public_local, R.drawable.ic_local_24dp, { TimelineFragment.newInstance(TimelineFragment.Kind.PUBLIC_LOCAL) })
+        FEDERATED -> TabData(FEDERATED, R.string.title_public_federated, R.drawable.ic_public_24dp, { TimelineFragment.newInstance(TimelineFragment.Kind.PUBLIC_FEDERATED) })
+        DIRECT -> TabData(DIRECT, R.string.title_direct_messages, R.drawable.reblog_direct_dark, { ConversationsFragment.newInstance() })
+        HASHTAG -> TabData(HASHTAG, R.string.hashtag, R.drawable.ic_hashtag, { args -> TimelineFragment.newInstance(TimelineFragment.Kind.TAG, args.getOrNull(0).orEmpty()) }, arguments)
```

Comment:
no idea how to get a reference to the `arguments` member inside the `fragment` function so I made a workaround by passing it into the function later

Reply:
There's only one way I can think of and it is `TabData.() -> Fragment` as declaration of a function.

Reply:
Or passing a whole `TabData` to the function, compiles to the same code but maybe is more straightforward, I'm not sure, maybe it's okay like it is now.

Reply:
I will leave it as it is, its not overly complicated